pkg/server/api/rpccontext/ratelimit.go

```diff
@@ -0,0 +1,28 @@
+package rpccontext
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type rateLimiterKey struct{}
+
+type RateLimiter struct {
+	Used bool
```

Comment:
The intention with `Used` is that the middleware layer will ensure that the rate limiter was used when the RPC has finished so we can detect when the server doesn't call it for a given RPC.